### ISSUE TITLE
 indel (SO:1000032) : Definition problem

### ISSUE THREAD

**Issue by sabrinatoro** [The-Sequence-Ontology/SO-Ontologies#445]
In the course of discussions reported in ticket #444 , we became aware that the current SO definition of “indel” (SO:1000032) is contrary to standard usage in our genetic research communities:

-Current definition: A sequence alteration which included an insertion and a deletion, affecting 2 or more bases.
-Previous definition: A hybrid term (insertion/deletion) to describe sequence length change when the direction of the change is unspecified.

On the MISO site, this change appears between release 2.4 (http://www.sequenceontology.org/browser/release_2.4/term/SO:1000032) and 2.5 (http://www.sequenceontology.org/browser/release_2.5/term/SO:1000032 ).  

The update introduces a key error (wrong conjunction); restricting to >2bp is also incorrect. 

Minimally, to bring the definition into line with common usage: 
SO:1000032 A sequence alteration which included an insertion **OR** a deletion.

Or more informative:
SO:1000032 A sequence alteration which included an insertion **OR** a deletion. Describes a sequence length change when the direction of the change is unspecified or when such changes are pooled into one category.

**Comment by keilbeck** [comment]
Hi Sabrina
Do you have a reference for when the definition of indel changed?

When I look at other resources, I see that the meaning in the SO definition is preserved - for example MeSH
https://www.ncbi.nlm.nih.gov/mesh?Db=mesh&term=INDEL+Mutation

ATG -->AC would bean indel in the traditional sense
In my mind INDEL is the term you use when an insertion and a deletion have both happened, but the order of operations is not clear. 

I think we need more context here as I may be old fashioned but I would choose AND rather than OR.

**Comment by sabrinatoro** [comment]
Hi Karen,

Yes indeed, it appears that the definition is preserved in MeSH (maybe they use an old version of SO?); however, the definition has been changed at SO itself based on the latest SO version. 
We do not have a reference for when/why the definition of indel changed. Based on the MISO site, it appears that this change was made between release 2.4 and 2.5. 

The current definition imply that both an insertion AND a deletion occurred, like in the following examples where x nucleotides are deleted, and y nucleotides are inserted. 
![indel](https://user-images.githubusercontent.com/12737987/45126634-370a5680-b129-11e8-99aa-ffcd1327e0b4.jpg)

However, the term "indel" historically means that there was either an insertion OR a deletion (not both), but we don't know which one it is. 

These 2 definitions (the old one meaning "or", and the new one meaning "and") refer to 2 different kind of variants, and therefore this change has a significant effect on the way this term has been used. 
(maybe we can also discuss this in the context of our discussion of case #444 )

**Comment by diekhans** [comment]
INDEL in genomics and evolutionary biology where there are differences in the sequence length where the exact order of events that lead to this differences is not known.  It may be an insertion in one, a deletion in another, or even a combination.

  https://www.sciencedirect.com/topics/biochemistry-genetics-and-molecular-biology/indel

While the MESH description is a bit odd in saying `blended', the detailed description matches the  evolutionary biology description:
   https://www.ncbi.nlm.nih.gov/mesh?Db=mesh&term=INDEL+Mutation

Please restore the old definition, the new one is not common usage.

**Comment by diekhans** [comment]
it seems that HGVS might have caused this dual meaning mess with http://varnomen.hgvs.org/recommendations/DNA/variant/delins/
If there really are two definitions being used, then INDEL it would be better if INDEL wasn;'t a SO term at all.

**Comment by keilbeck** [comment]
@fcunningham Fiona, could you weigh in to this thread? Cheers.

Given a reference genome and  comparison genome, I am not sure how you would not know whether a deletion or an insertion happened if it is a simple insertion or deletion. If you look at the root node here:
sequence_alteration
A sequence_alteration is a sequence_feature whose extent is the deviation from another sequence.
The comparison is always from a reference to another sequence.  So there is no need for a term that is 'either deletion or insertion'

For the genomic variant annotation community, it seems like SO and HGVS should be in agreement over the term INDEL. "An "indel", deletion/insertion in HGVS nomenclature, is a variant which is a combination of a deletion and an insertion. Based on existing nomenclature, the variant can be described as a deletion and insertion occurring at the same position, using the format g.112_117delinsTG."

Removing INDEL from SO is not an option, as the term is used in annotations in the same way it is used by HGVS.

We could add another term to satisfy your use-case.

**Comment by diekhans** [comment]
One does not know if an evolutionary change is an insert or a deletion without the outgroup, which has often been the case.  When known, it is referred to as either an insertion or a deletion relative to the other species, not as an INDEL.  Although casually, INDEL often still gets used.  
  
I have contacted HGVS about this.  Their documentation is very confusing where they call this a `delins' is most cases, but an INDEL in others.  I don't know any evolutionary genomicists that use 
INDEL this way.  In the formal part of the defintion, the deinition is not rigrious.

   “delins” = type of change is a deletion-insertion (indel) = delins

This isn't just SO, we need to put a stop to this confusion or the clinical and research will have even more trouble communicating.  Please help!  We can change decades of definition just because the HGVS specication is poorly written.

Are there any other citations besides HVGS that use this definition?  A new term leaving in INDEL will not help, as the majory of users will still use the traditional meaning of INDEL.

**Comment by murphyte** [comment]
I agree with Mark on this. Wikipedia and elsewhere define indel as "insertion or deletion":
https://en.wikipedia.org/wiki/Indel

I don't see how SO can use a different definition and not risk having a substantial fraction of users misinterpret the term because they're assuming they know what it means and not look it up. I always thought that was the motivation behind the 'delins' term in HGVS, to be obviously different from indel.

**Comment by diekhans** [comment]
From looking at the document, it appears they tried to change indel to delins and just missed some of the places.  Even a link to an indel.html page is broke, because the page has been renamed insdel.html.

The HGVS document has never been a rigorous specification, which has cause all type of problems attempting to parse HGVS specification.  In there defense, they get zero money for doing this work.

Both HGVS and SO to make corrections.

**Comment by keilbeck** [comment]
Mark
The whole purpose of the SO is to do genome annotation. For the sequence alterations there has to be a reference or none of the terms make any sense. Something is an insertion relative to something else. One sequences insertion is another sequences deletion. Having a SO term be an unknown for whether it is an insertion does not fit with the logic.

The definition of INDEL is not confined to just SO or HGVS.
Annotation tools have used this to mark up genomic sequence. 
There is legacy usage of these terms.

I am not comfortable making a  huge change to the meaning of a popular term without consensus from the genomic annotation community and without a plan to disseminate the change. At this point I am unsure if the proposed changes help or hinder. 
I also don't know who is part of your collaboration and who is missing. Who has not had input? I would like to hear from the EBI on their interpretation.

**Comment by diekhans** [comment]
Hi Karen,


While it is desirable to be able to unambiguously annotate genomes, this is not always possible given errors in the assemblies, uncertainty in the evidence, and uncertainty in the biology.  Instead of making potentially incorrect guesses, concepts like ambiguous insertions/deletions are important for correct annotation.  A given genome being a "reference" is not relevant, as we will be dealing with understanding evolutionary history using thousands of reference assemblies.

Would you be so kind as to point me at tools that using the "new" definition of INDEL?   At this point, I am not convinced that this redefinition was even intended by HGVS.

I completely agree that any change should be well announced and vetted.  I am somewhat amazed to find that INDEL has been redefined without viability in the genome annotation communities.  The change to SO is far less urgent than getting the confusion under control.

While I am speaking only for myself, I am part of the GENCODE consortium and the effort to adopt SO to replace the bespoke controlled vocabulary.  I am also part of the Genome 10K Vertebrate Annotation Project, Bird 10K, and 200 mammals.  So I can provide a conduit to these groups, which are a huge portion of the current vertebrate genome annotation projects.

I will be meeting with EBI, NCBI, and many others at the Genome 10K meeting next week and discussing this issue.  Please attend if you can.  Please mail me at markd@ucsc.edu and we can arrange to discuss this with a larger audience.

Mark

**Comment by diekhans** [comment]
HGVS fixed the broken link.  Looking at 
    http://varnomen.hgvs.org/recommendations/protein/variant/delins/

The QA no long describes delins as being an INDEL, however, it does casually mention it in the next paragraph.  This can't be considered a normative definition.

**Comment by keilbeck** [comment]
Mark
I have reached out to two other institutes for feedback about your use-case and proposal, and am waiting for them to compose their thoughts.

I am not able to attend the 10K meeting in New York this week given the short notice and my teaching schedule this semester.

I think your work with these large vertebrate genome projects represents a new use-case for SO to handle. As mentioned before - the logic behind SO sequence_alterations is built upon the idea of comparison, between a known and a new. The driving use-case for this development has been precision medicine.   So there probably is not a quick fix but will need more discussion from multiple view points.

While HGVS is is clunky and sometimes hard to parse, it is the de facto standard for clinical representation of variants at the present time. There is much annotation in existence that relies on HGVS, and the annotation of indels as defined currently in SO.

I see this as an opportunity to come together and figure out how to do the right thing for all parties. I think this may be a slower process than you hoped though.

--K

**Comment by thefferon** [comment]
My name is Tim Hefferon; I am the lead curator at dbVar (NCBI's database of structural variation). I have long been bothered by the divergent usage of 'indel' and the confusion that causes; I would like to see the issue resolved. I agree with Mark that the confusion needs to be contained; however I do not necessarily agree that the best way to do that is to rush to update SO:1000032 (though it may be; and there may be better ways to do so than others).

I strongly agree (with Karen, I think, though I don't want to put words in her mouth) that the whole point of SO is to eliminate existing ambiguities by specifying terms and definitions; and that doing so in the context of genetic variations fundamentally requires one to posit a reference-to-alternate relationship between sequences (something that is notably and intentionally absent from the evolutionary genomics point of view – at least as I understand it).

I propose there are three main usages, or interpretations, of the term 'indel'. (I also propose that everyone has a natural tendency to think of that usage or interpretation with which they are most familiar, as the one that is "preferred", "best", or "most commonly understood".) They are (and please set me straight if you think I am wrong):

1. sdkfgjhdfg

**Comment by keilbeck** [comment]
We can definitely add a footnote to the term while this is under discussion.
I am excited to be part of the VMC and equally hope that it removes the ambiguity.

**Comment by thefferon** [comment]
Hi All,

My name is Tim Hefferon; I am the lead curator at dbVar (NCBI's database of structural variation). I have long been bothered by the divergent usage of 'indel' and the confusion it causes; I would very much like to see the issue resolved. I agree with Mark that the confusion needs to be contained; however I do not necessarily agree that the best way to do that is to rush to update SO:1000032 (though it may be – and there may be different ways to do so).

I strongly agree (with Karen, I think, though I don't want to put words in her mouth) that the whole point of SO is to eliminate existing ambiguities by specifying terms and definitions. Accomplishing this in the context of genetic variation requires fundamentally that one posit a reference-to-alternate relationship between the sequences (this feature that is notably and purposefully absent from the evolutionary genomics point of view – at least as I understand it).

I propose three main usages, or interpretations, of the term 'indel'. (Everyone has a natural tendency to think of the usage or interpretation with which they are most familiar as the one that is "preferred", "best", or "most commonly understood" – myself included.) Please set me straight if you think the following are incorrect:

INDEL:

**1a.** a specific type of variant in which one sequence is replaced by a different sequence (of the same or different length)

**1b.** (pedantic version) a specific type of variant in which one or more nucleotides are replaced by two or more different nucleotides
Synonyms: ‘insertion-deletion’, ‘deletion-insertion’, 'delins', 'multinucleotide substitution'

**2.** an ambiguous catch-all term used to refer collectively to a general class of variant that includes all small insertions and deletions (with no indication as to size)

**3.** A sequence alteration in which a contiguous sequence is present in one sample or outgroup, but absent in another (or "in others"). (Such a variant **could** specifically be called an insertion or a deletion, **if** a reference-to-alternate relationship were asserted, but without a reference it's just a "difference". (Isn't this what is usually called a Presence-Absence Variant?)

Both 1 and 2 imply the existence of a reference-to-alternate relationship between the compared sequences.

Above, Mark said, 

> Minimally, to bring the definition into line with common usage: SO:1000032 A sequence alteration which included an insertion OR a deletion.

I don't dispute that this concept might be useful in some contexts, but – since the primary purpose of the SO term and definition is specificity – I don't think a definition that includes ambiguity has a place in SO. Furthermore, no term has much hope of serving the purpose of specificity unless a reference-to-alternate relationship is asserted.

**Bottom line for me:** I used to hold out hope of convincing "the community" – i.e. everyone – to accept definition #1 as the default. However in face of the pervasiveness of definition #2, I think it may be best if we were to begin referring to variants matching definition #1 – as HGVS appears to have done – by another term, perhaps '**delins**', '**deletion-insertion**', or '**multinucleotide substitution**'.

My apologies if I have missed the point along the way, or mis- or under-represented a particular position.

–Tim

**Comment by thefferon** [comment]
> INDEL in genomics and evolutionary biology where there are differences in the sequence length where the exact order of events that lead to this differences is not known. It may be an insertion in one, a deletion in another, or even a combination.
> 
> https://www.sciencedirect.com/topics/biochemistry-genetics-and-molecular-biology/indel

As agnostic food for thought, here are the definitions from the first few sources in Mark's link above:

---

**Indel**
An indel is an insertion or deletion polymorphism – a sequence of nucleotides are added or deleted in some individuals creating a polymorphism at that site.

_– From: Philosophy of Biology, 2007_

---

**Indels** are the second most common class of mutation in the human genome (). They consist of an insertion or deletion of one or more DNA bases into a genome. This can have far-ranging effects concerning gene expression and genetic disease (). Detecting and identifying indels is a multistep process in which error can be introduced at every step. Starting with a set of DNA sequence reads and a reference DNA genome, reads are first mapped to the reference genome with a mapping program and then the mapped results are inputted into indel-calling software to identify indels.

_– A Comparative Analysis of Read Mapping and Indel Calling Pipelines for Next-Generation Sequencing Data; Jacob Porter, ... Liqing Zhang, in Emerging Trends in Computational Biology, Bioinformatics, and Systems Biology, 2015_

---

**Indel** Definition and Relationship to Other Classes of Mutations
“**Indel**” is a general term that may refer to insertion, deletion, or insertion and deletion of nucleotides in genomic DNA.

_– Insertions and Deletions (Indels)
Jennifer K. Sehn, in Clinical Genomics, 2015_

---

Insertions and Deletions
Insertions and deletions (**indels**) are when DNA is lost (deletion) or gained (insertion) on a smaller scale, defined as <1000 bp (Fig. 1.4D). If the indel occurs within a coding region, it is described as being “in-frame” if the amount of DNA lost or gained is divisible by 3; alternately, it is called a “frameshift” because the triplet reading code is altered for all subsequent nucleotides.

_– The Newly Emerging View of the Genome
Stephan J. Sanders, Christopher E. Mason, in Genomics, Circuits, and Pathways in Clinical Neuropsychiatry, 2016_

**Comment by diekhans** [comment]
Thanks @thefferon 

At least one source I greatly trust (clinical genomics, extensive HGVS expertise) was under the impression that INDEL and delins are synonymous.   Others (cancer genomics, comparative genomes, ancient DNA),  held the traditional view and were perplexed by the new definition.

Given there is both confusion and a long history of the term, it seems better to drop it for other terminology.  INDEL is rather slang-like.

There also is an issue with the historic use of SO:1000032 under the old definition, as the new one is fundamentally different. 

The other issue of having a term for annotating a difference between two sequences when there is uncertain history seems more complex and subtle and should be considered separately.   I think the breaks down into the use cases of observed sequence differences vs putative evolutionary history.

In comparative genomics, claiming a certain evolutionary history when there is no data to support is less precise than stating it is ambiguous.    Although, maybe the scope of SO doesn't include annotating evolution history.  My reading of the goals statement on the home page appear to put both comparative and clinical genomics out-of-scope (of course, goals can change).

**Comment by cmungall** [comment]
I have only skimmed the discussion, but a general OBO-eque SOP for dealing with these kinds of things (Karen knows this already but not sure how familiar others are, this is all consistent with what Mark is saying)

 1. Obsolete the ambiguous/problematic class (e.g. SO:1000032)
 2. Create N new classes for each of the distinct formally defined concepts. Each of these will have a new ID, and a reasonably formal definition
 3. Include metadata such that SO:1000032 points to the N new classes, which assists in manual migration
 4. Give each of the N new classes an ugly but unambiguous primary label
 5. Add as many synonyms as you like, and add metadata to the synonyms. The synonym "indel" could be shared by >1 class, so long as it is not scoped as an exact synonym

Before actually doing this there would be an announcement giving people instructions on how to migrate their existing SO assignments.

There are tradeoffs here. Obsoletion causes a lot of churn. For SO in particular, it can be hard to go back to retrospective GFFs or whatever and manually migrate to the appropriate one of the N new terms, due to the lack of a centralized database. But this churn has to be balanced against the cost of having inconsistent usage of the class.

More to say on the actual genomics aspect of this later...

**Comment by diekhans** [comment]
At the G10K meeting I chatted with senior staff members of NCBI and Ensembl annotation groups, as well as GRC group.  They were all perplexed by the change in the definition of `INDEL' given the history of genome annotation.

I believe this is all very resolvable if we put the right group of people together.

**Comment by sarahhunt** [comment]
Hi All,

In Ensembl, we use 'indel' to describe substitutions of unequal length such as AA/TTTT. We don't use it as grouping term for insertions and deletions - we always describe changes relative to a reference if possible, so have a direction and use the precise term.

If the term 'indel' has has both the 'length changing substitution' meaning and 'insertion or deletion' meaning, would it make sense to describe it as a sequence alteration where the sequence length is altered due to zero or more bases being deleted and zero or more being inserted? This would make it a parent term for insertions and deletion (so maintain the grouping function) and keep the 'length changing substitution' meaning.

There are few variants described as indels in our databases so there would not be a huge impact in re-classifying these if that is preferable.

We didn't expect the meaning of SO terms to change - is there any system for announcing such changes? We don't want to end up with a situation where the logic of our assignment code does not match the new description of the term we have assigned.

Sarah

**Comment by thefferon** [comment]
I like Sarah's explanation. However I have never understood the basis of the need for the lenth-changing aspect of multinucelotide substitutions to play such a central role in the differentiation of variants. I would like to see this issue fleshed out - especially if it is going to play such a central role in structuring terms within SO.

Consider the following example of a reference followed by three variants, in which 8 bases of the reference have been replaced by 6, 8, and 10 bases, respectively:

| | sequence | length | delta |
|:--------:|:--------:|:--------:|:--------:|
|REF|CCTAG--CGTCTAGC--TTCGA|18 bp|
|VAR A|CCTAG---TCTAGT---TTCGA|16 bp|-2|
|VAR B|CCTAG--ACTCTAGG--TTCGA|18 bp|0|
|VAR C|CCTAG-CTTAGCTATG-TTCGA|20 bp|+2|

It seems to make more sense to me to group these variants together and call them "deletion-insertions" or better - "multinucleotide substitutions" - rather than separate out variant 'B' because it happens not to change the overall allele length (this observation is limited to non-coding DNA of course, where reading frame is not an issue).

If the emphasis on whether a variant is length-changing originates mainly from discussions of coding sequence variants, or of very short-length variants, someone please explain, in those cases, the justification for it to be so significant a differentiator.

**Comment by diekhans** [comment]
Nice summary, some thoughts:

1.   Definitions of terms should not change, per @cmungall there are ways for an ontology to handle this.  So HGVS delins would get a new SO id, not be a change in to SO:1000032. 
2. Our disagreement on INDEL is really a divergence of the meaning of the term.  While I am happy to try to prove my meaning was first, that really doesn't help matters,  It is painfully obvious, that INDEL has become confused.  Thus, there should not be a primary term INDEL. Possibly a new name for SO:1000032, or obsoleting SO:1000032.  
3. I think the issue of relative to a reference is more a matter of there being two use cases and the meaning isn't exactly clear.  Some of this confusion might be due to reference as in direction of comparison vs evolutionary comparison.  Is this an observed change in the reference vs a difference in two sequences with putative evolution labeling?  There are some interesting, subtle issues here that need some through.   I suspect both cases can be handled, as it is the context were the SO term is used that differentiates the two cases.

**Comment by sabrinatoro** [comment]
Good ontological practice requires that SO:1000032 be obsoleted. However a large number of genetic research communities have been using the term “indel” for decades -- they are not going to stop doing so, just because there was a mistake in SO. 
Maybe the original definition (insertion OR deletion) should be restored OR (if SO:1000032 is obsoleted) a new term called ‘indel’ should be created (with the original definition). Either way, it would be good to include in the description of 'indel' a caution that it is an imprecise term and that for the purposes of most users of SO a more precise term is the better choice

We do need another term to express a sequence alteration which includes multiple changes within a contiguous region relative to a reference sequence. This term would encompass the cases when there is an insertion+deletion with (or without) change in length compared to a sequence of reference (see ticket #444)

**Comment by keilbeck** [comment]
Sabrina
When the definition of indel changed several years ago - we should have obsoleted and started again. That did not happen, it was an oversight.  But things have moved on. The large databases today use the definition in SO as is today, with the meaning more or less in line with HGVS. In this respect, we can update the name, keep the definition and the ID. We will not reinstate the old definition. We are offering to add a new term to fill that gap for your team, but we have the caveat that the definition insertionORdeletion  does not fit within the  overall definition of a sequence alteration being a comparison against a reference - where you always know the direction, if it is an insertion or deletion. We will add indel as a synonym to your term. 

We are asking you to look at the term complex substitution, to see if it fits the concept that you need for  #444.

**Comment by diekhans** [comment]
@keilbeck, would you please point us at some of the large databases using the `new' version of indel?  It would be helpful to have the context.

Another suggestion is `unspecifed_indel'.

thanks!

**Comment by keilbeck** [comment]
see message from Sarah at ebi with regards to current usage

unspecified_indel sounds perfect for what you need

**Comment by sabrinatoro** [comment]
@keilbeck :
When my team looked into SO terms, we discovered the issue about SO:1000032 definition and reported it (this entire ticket). 
However, the term that we need is to express a sequence alteration in comparison against a reference, since in our case we ALWAYS know what the reference is. 
We thought that SO:1000032 would either be obsoleted or the old definition would be reinstated. This is why we created ticket #444 to request a new term.

SO:1000032's current definition is basically what we need. Therefore if the term name is changed (to avoid confusion) and the ID and definition are maintained, I believe we could use this term and ticket #444 can be closed (I will check with my team to make sure).

Groups who have been using "indel" in its original meaning (i.e. insertion OR deletion) should weight in on which term&definition would be appropriate to use.

**Comment by sabrinatoro** [comment]
@keilbeck 
I wanted to check the status of this ticket (we need to decide which term to use). 
Based on previous comments, it looks like the decision is (or would be) to rename SO:1000032 and keep the current definition. Is this the final decision? or are discussions still taking place? 

If it is the final decision, 
1) what would the new name be? delins as in HGVS?
2) when can we expect SO to be updated with this change? 

Thank you very much!

**Comment by diekhans** [comment]
I believe it is to make SO:1000032 obsolete with the current name, and create two new entries link to it.  One with the HGV definition (delins) an another that is unspecified_indel,  It would be good to document SO:1000032 to explain the error in updating SO.

**Comment by sabrinatoro** [comment]
Is there any timeline on when these changes in the ontology will take place? Thanks

**Comment by nicoleruiz** [comment]
This is Nicole, one of Karen's grad students. I was under the impression that she wanted to set up a call with everyone to discuss the various options to resolve this ticket. 

@keilbeck Do you want to set up a time for everyone to discuss this?

**Comment by diekhans** [comment]
@sabrinatoro I would be happy to work on a pull request with you as a proposal, although I don't know the mechanism for update SO, so would needed some hand holding.

@keilbeck, I am at EBI through Jan 14th, so could facilitate a call with the appropriate EBI people.

**Comment by cmungall** [comment]
I made a separate ticket for creating an SOP for PRs: #463

**Comment by nicoleruiz** [comment]
I have made the following changes:
- Changed SO:1000032 term name indel to delins and made indel a synonym
- Created new term SO:0002217 unspecified_indel with definition "A sequence alteration which includes an insertion or a deletion. This describes a sequence length change when the direction of the change is unspecified or when such changes are pooled into one category."

**Comment by davidwsant** [comment]
Hi guys,

It looks like @nicoleruiz has made the updates, but the issue was never officially closed out. I am going to close it now, but if some of the issues have not been resolved thoroughly you are welcome to open another issue request.

Thanks,

Dave
